### DISCUSSION TITLE
Example on how to implement bar chart with timestamps as Y axis?

### DISCUSSION THREAD

**Discussion by koyuul** [nasa/openmct#7869] [Community Q&A] [answered]
Hello!

I am attempting to implement a bar chart view of data that I have been able to successfully graph using the standard line graphs. With the x-values being integers and y-value being Dates. 

However, when creating the plot it doesn't seem to pick up on the y-axis. When dropping the value into the bar chart similar to how I'd drop the value into an overlay chart, I was expecting to get the bar chart, with it being split into days (?) 

Overall, I'm generally unsure of how the progress of using the bar charts work. I have tried searching for any documentation but I could only find pdf's from 2015 with what looks like outdated tutorials. Could anyone point me in the right direction for utilizing bar charts?

The domainObject I am trying to drop in has the measurement formatted as:
```
       {
            "name": "generated_PACKET_COUNT",
            "key": "generated_PACKET_COUNT",
            "values": [
                {
                    "key": "value",
                    "name": "generated_PACKET_COUNT",
                    "units": "packets per day",
                    "format": "int",
                    "hints": {
                        "range": 1
                    }
                },
                {
                    "key": "utc",
                    "source": "timestamp",
                    "name": "Timestamp",
                    "format": "utc",
                    "hints": {
                        "domain": 1
                    }
                }
            ]
        }
```

The current output looks as such
![Screenshot from 2024-10-04 12-26-54](https://github.com/user-attachments/assets/70aeee5d-dcc7-42b8-b935-45d78167d92d)

Any help is appreciated -- thank you!

**Comment by shefalijoshi** [comment, marked as answer]
Hi @koyuul 

Bar charts exclude 'time' (domain hints) metadata specifically from the x and y axis values. Their intention is to only plot range values.
Once way you could accomplish plotting packets over a range of days is by adding metadata values like:
```
{
        key: 'days',
        name: 'Days',
        unit: 'day',
        format: 'integer[]', //(or string[])
        hints: {
          range: 3
        }
 },
{
        key: 'packets',
        name: 'Days',
        unit: 'day',
        format: 'integer[]', //(or string[])
        hints: {
          range: 4
        }
 }
```

Then your telemetry would need to return data in arrays like:
```
days: [1743619821, 1743706221, 1743792621]
packets: [30, 80, 10]
```

Once you do this, you can Edit your bar chart and choose which metadata you want to display on which axis, by going to the inspector `Config` tab.

Let us know if we can help you further.
